libs/blocks/global-navigation/utilities/utilities.js

```diff
@@ -95,13 +96,26 @@ export const getFederatedUrl = (url = '') => {
   return url;
 };
 
+const getPath = (urlOrPath = '') => {
+  try {
+    const url = new URL(urlOrPath);
+    return url.pathname;
+  } catch (error) {
+    return urlOrPath.replace(/^\.?\//, '/');
+  }
+};
+
 export const federatePictureSources = (section) => {
-  section?.querySelectorAll('[src], [srcset]').forEach((source) => {
-    const type = source.hasAttribute('src') ? 'src' : 'srcset';
-    const value = source.getAttribute(type);
-    if (!value) return;
-    source.setAttribute(type, value.replace(/^\.?\//, `${getFederatedContentRoot()}/`));
-  });
+  section?.querySelectorAll(`[src*="/${FEDERAL_PATH_KEY}/"], [srcset*="/${FEDERAL_PATH_KEY}/"]`)
+    .forEach((source) => {
+      const type = source.hasAttribute('src') ? 'src' : 'srcset';
+      const path = getPath(source.getAttribute(type));
+      const localeOrFederalKey = path.split('/')[1];
+      const federalKey = path.split('/')[2];
+      const isFederal = localeOrFederalKey === FEDERAL_PATH_KEY || federalKey === FEDERAL_PATH_KEY;
+      if (!isFederal) return;
```

Comment:
Given the query selector used, we're certain that `/federal/` is present in the path, why is this extra logic required?

Reply:
It's based on a comment from [WebPS](https://jira.corp.adobe.com/browse/WPS-24098?focusedCommentId=39050693&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-39050693) to avoid matching URLs that have /federal/ anywhere in the path, such as `/acrobat/programs/federal/government.svg` (non existant example url)

Reply:
Then let's trim this a bit, something like `[, first, second] = path.split('/'); if ([first, second].includes(FEDERAL_PATH_KEY)) return;` (silly variable naming, use whatever you see fit)

Reply:
Nice suggestion, done!